docs/user/recipes/raw-url-path.rst

```diff
@@ -0,0 +1,110 @@
+.. _raw_url_path_recipe:
+
+Decoding Raw URL Path
+=====================
+
+This recipe demonstrates how to access the "raw" request path using
+non-standard (WSGI) or optional (ASGI) application server extensions.
+This is useful when, for instance, a URI field has been percent-encoded in
+order to distinguish between forward slashes inside the field's value, and
+slashes used to separate fields. See also: :ref:`routing_encoded_slashes`
+
+WSGI
+----
+
+In the WSGI flavor of the framework, :attr:`req.path <falcon.Request.path>` is
+based on the ``PATH_INFO`` CGI variable, which is already presented
+percent-decoded. Some application servers expose the raw URL under another,
+non-standard, CGI variable name. Let us implement a middleware component that
+understands two such extensions, ``RAW_URI`` (Gunicorn, Werkzeug's dev server)
+and ``REQUEST_URI`` (uWSGI, Waitress, Werkzeug's dev server), and replaces
+``req.path`` with a value extracted from the raw URL:
+
+.. code:: python
+
+    import falcon
+
+
+    class RawPathComponent:
+        def process_request(self, req, resp):
+            raw_uri = req.env.get('RAW_URI', req.env.get('REQUEST_URI'))
```

Comment:
maybe we could the uglier 
```
raw_uri = req.env.get('RAW_URI')
if not raw_uri:
   raw_uri  = req.env.get('REQUEST_URI')
```
since it should be faster in when finding `RAW_URI`

Reply:
maybe just a note above is enough. the if form is not pretty

Reply:
Heh, I wasn't trying to optimize recipes _that_ much.

Maybe `or` could be more performant?
```python
            raw_uri = req.env.get('RAW_URI') or req.env.get('REQUEST_URI')
```

Reply:
seems the best option 👍